docs/notebooks/external_callbacks.md

```diff
@@ -0,0 +1,159 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: Python 3
+  name: python3
+---
+
++++ {"id": "7XNMxdTwURqI"}
+
+# External Callbacks in JAX
+
++++ {"id": "h6lXo6bSUYGq"}
+
+This guide is a work-in-progress outlining the uses of various callback functions, which allow JAX code to execute certain commands on the host, even while running under `jit`, `vmap`, `grad`, or another transformation.
+
+This is a work-in-progress, and will be updated soon.
+
+*TODO(jakevdp, sharadmv): fill-in some simple examples of {func}`jax.pure_callback`, {func}`jax.debug.callback`, {func}`jax.debug.print`, and others.*
+
++++ {"id": "dF7hoWGQUneJ"}
+
+## Example: `pure_callback` with `custom_jvp`
+
+One powerful way to take advantage of pure callbacks is to combine them with `custom_jvp` (see [Custom derivative rules](https://jax.readthedocs.io/en/latest/notebooks/Custom_derivative_rules_for_Python_code.html) for more details on `custom_jvp`). Suppose we want to create a JAX-compatible wrapper for a scipy or numpy function that is not yet available in the `jax.scipy` or `jax.numpy` wrappers.
```

Comment:
I think it might be worth linking the pure_callback docs here (https://jax.readthedocs.io/en/latest/_autosummary/jax.pure_callback.html)

Reply:
Done